### PR TITLE
[topgen] Inter-module validation code

### PR DESCRIPTION
**Pull request by eunchan**
topgen didn't have a validation code for inter-module signal.
This commit is to implement basic signal format part for inter_module.
It checks below:

1. Check if req/rsp format is `module.signal{[index]}` ({} means
   optional)
2. Check if request is array and matched the number of entries
3. Check if response has an array index, and request is not an array

### CODE REVIEW

File: util/topgen/intermodule.py [hunk 1]
```diff
@@ -146,9 +187,121 @@ def find_intermodule_signal(sig_list, m_name, s_name) -> Dict:
...
+        return 0
+
+    total_error = 0
+
+    for req, rsps in topcfg["inter_module"].items():
+        error = 0
+        # checking the key, value are in correct format
+        # Allowed format
+        #   1. module.signal
+        #   2. module.signal[index]
```
Comment (tjaychen) [inline util/topgen/intermodule.py:205]: could you provide examples for each of these cases? i feel like that would be easier to digest to a reader.. or more likely..someone who did something wrong and is trying to figure out why :)

**Comment by tjaychen** [comment]
i think it looks good, but i'd really appreciate an example or two (see comment), i can make sure i'm reading it right.

File: util/topgen/intermodule.py [hunk 2]
```diff
@@ -138,6 +157,28 @@ def elab_intermodule(topcfg: OrderedDict):
...
+def filter_index(signame: str) -> Tuple[str, str, int]:
+    """If the signal has array indicator `[N]` then split and return name and
+    array index. If not, array index is -1.
+
+    param signame module.sig{[N]}
+
+    result (module_name, signal_name, array_index)
+    """
+    pattern = re.compile('(\w+)\.(\w+)(\[(\d+)\])*')
+    m = pattern.match(signame)
```
[the remark this answers is not in the document]
Comment (eunchan) [reply util/topgen/intermodule.py:169]: Right.. `match` can be directly used ;)

File: util/topgen/intermodule.py (hunk 1 above)
Comment (eunchan) [reply util/topgen/intermodule.py:205]: OK I will revise the code to have more examples there.
Also the missing part in this PR is that updating the docs.
Comportability spec and topgen manual page need to be revised to have `inter_signal_list` and `inter_module` fields.

**Pull request merged**